configure

```diff
@@ -128,34 +128,47 @@ for (var f of images) {
     .using('copy')
 }
 
-let boardFolders = globule.find('boards/*/*/*', {filter: 'isDirectory'})
+const repositories = globule.find('boards/*/*/*', {filter: 'isDirectory'})
+const boards = repositories.reduce((newBoards, folder) => {
+  const [info, yamlPath] = getKitspaceYaml(folder)
+  if (info.multi) {
+    for (let board in info.multi) {
+      const boardInfo = info.multi[board]
+      boardInfo.multiKey = board
+      boardInfo.yamlPath = yamlPath
+      boardInfo.repoPath = folder
+      boardInfo.boardPath = path.join(folder, board)
+      newBoards.push(boardInfo)
+    }
+  } else {
+    info.yamlPath = yamlPath
+    info.repoPath = folder
+    info.boardPath = folder
+    newBoards.push(info)
+  }
+  return newBoards
+}, [])
+
+function getKitspaceYaml(folder) {
+  let file, filePath
 
-boardFolders = boardFolders.reduce((newBoardFolders, folder, index) => {
-  let info
-  let file
   if (fs.existsSync(`${folder}/kitnic.yaml`)) {
     file = fs.readFileSync(`${folder}/kitnic.yaml`)
+    filePath = `${folder}/kitnic.yaml`
   } else if (fs.existsSync(`${folder}/kitspace.yaml`)) {
     file = fs.readFileSync(`${folder}/kitspace.yaml`)
+    filePath = `${folder}/kitspace.yaml`
   } else if (fs.existsSync(`${folder}/kitspace.yml`)) {
     file = fs.readFileSync(`${folder}/kitspace.yml`)
+    filePath = `${folder}/kitspace.yml`
   }
   if (file != null) {
-    info = yaml.safeLoad(file)
-  } else {
-    info = {}
+    const info = yaml.safeLoad(file)
+    return [info, filePath]
   }
 
-  if (info.multi) {
-    for (let project in info.multi) {
-      projectPath = path.join(folder, project)
-      newBoardFolders.push(projectPath)
-    }
-  } else {
-    newBoardFolders.push(folder)
-  }
-  return newBoardFolders
-}, [])
+  return [{}, folder]
```

Comment:
Why are you returning a folder here when there is no yaml? I can see that that could cause issues in the build system.

Reply:
Yeah, good catch, that was part of my issue.

Originally added it so the `deps` array would always contain four valid paths in `processBOM`; which also lead to me adding the try/catch.

Re-arranging that `deps` array, and moved the `yamlPath` to the end. Let me get rid of the try/catch and just check for the existence of `yamlPath`.